### DISCUSSION TITLE
Private stored procedures

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [supabase/supabase#2224] [Questions] [answered]
Stored procedures seem to be created any time you create a function in Postgres. However, to use a trigger (e.g., create the public.users row when a new user signs up), you also need to create a function. And these functions seem to be callable from `supabase-js`.

Is there anyway to make some functions not callable as RPCs?

**Comment by steve-chavez** [comment, marked as answer]
Hey @Nick-Mazuk,


All functions in the `public` schema will be exposed through RPC. To keep them private, you can create them on another schema. Like:

```sql
create schema private; -- can be any name
create or replace function private.my_func(..)
```
